src/components/App.js

```diff
@@ -197,52 +173,27 @@ const App = () => {
     useEffect(() => {
         if (didPostLoadEffects) return;
         (async () => {
-            await dispatch(fetchOpenIdConfigurationIfNeeded());
+            await dispatch(fetchOpenIdConfiguration(OPENID_CONFIG_URL));
             await dispatch(fetchUserDependentData(createEventRelayConnectionIfNecessary));
             setDidPostLoadEffects(true);
         })();
     }, [dispatch, createEventRelayConnectionIfNecessary, didPostLoadEffects]);
 
-    useEffect(() => {
-        // initialize session refresh worker
-        if (!sessionWorker.current) {
-            // Use session worker to send pings to refresh the token set even when the tab is inactive.
-            const sw = new SessionWorker();
-            sw.addEventListener("message", () => {
-                dispatch(refreshTokensIfPossible());
-                dispatch(fetchUserDependentData(nop));
-            });
-            sessionWorker.current = sw;
-        }
-    }, [sessionWorker]);
+    useSessionWorkerTokenRefresh(
+        CLIENT_ID,
+        sessionWorker,
+        createSessionWorker,
+        fetchUserDependentData(nop),
```

Comment:
this is going to result in constant re-running of the effect; it should be in a useMemo or since it doesnt depend on anything inside, factored out into a constant

Reply:
Removed nop since it was redundant anyway.

Reply:
I don't think it was - the action factory takes a function callback argument, and it won't work as an action itself - it needs to be passed something to work as a callback (this could probably be set as a default argument in the action factory instead, but either way `fetchUserDependentData` won't work since it's a factory, not an action)

Reply:
Sorry for my brain fart, I missed that `fetchUserDependentData` was a thunk action creator. 
You're right, I changed it in both places and the inner thunk function is in a constant now.